<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TestMainWindow</class>
 <widget class="QMainWindow" name="TestMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>11</pointsize>
    <weight>75</weight>
    <bold>true</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>TestMainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1401</width>
      <height>1081</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <property name="iconSize">
     <size>
      <width>20</width>
      <height>16</height>
     </size>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>连接</string>
     </attribute>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>311</width>
        <height>80</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>IP地址：</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>30</y>
        <width>341</width>
        <height>81</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="connect_button">
         <property name="text">
          <string>连接</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_2"/>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="getdata_btn">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>180</y>
        <width>121</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Agency FB</family>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>获取数据</string>
      </property>
     </widget>
     <widget class="QPushButton" name="freemode_btn">
      <property name="geometry">
       <rect>
        <x>340</x>
        <y>180</y>
        <width>131</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>自由拖动</string>
      </property>
     </widget>
     <widget class="QPushButton" name="close_btn">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>180</y>
        <width>151</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>关闭机器臂</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>350</y>
        <width>651</width>
        <height>87</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QTextEdit" name="pose0_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="pose1_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="pose2_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="pose3_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="pose4_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="pose5_txt"/>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>360</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>18</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>位姿：</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>460</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>18</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>速度：</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_4">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>450</y>
        <width>651</width>
        <height>87</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QTextEdit" name="speed0_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="speed1_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="speed2_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="speed3_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="speed4_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="speed5_txt"/>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>550</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>18</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>力：</string>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_5">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>540</y>
        <width>651</width>
        <height>87</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QTextEdit" name="force0_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="force1_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="force2_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="force3_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="force4_txt"/>
       </item>
       <item>
        <widget class="QTextEdit" name="force5_txt"/>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="halfballkeep_btn">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>260</y>
        <width>141</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>雕刻打磨</string>
      </property>
     </widget>
     <widget class="QPushButton" name="freedrive_btn">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>121</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>自带拖动</string>
      </property>
     </widget>
     <widget class="QPushButton" name="freedrive_close_btn">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>260</y>
        <width>171</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>自带拖动关闭</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>直线保持</string>
     </attribute>
     <widget class="QPushButton" name="KeepStraight_btn">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>160</y>
        <width>181</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>直线保持</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="line_pt1">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>170</y>
        <width>211</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="line_pt2">
      <property name="geometry">
       <rect>
        <x>733</x>
        <y>170</y>
        <width>211</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>90</y>
        <width>201</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>19</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>线段的两点：</string>
      </property>
     </widget>
     <widget class="QPushButton" name="close_btn_3">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>330</y>
        <width>171</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>关闭机器臂</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>平面保持</string>
     </attribute>
     <widget class="QPushButton" name="PlaneKeep_btn">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>171</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>平面保持</string>
      </property>
     </widget>
     <widget class="QPushButton" name="close_btn_2">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>130</y>
        <width>171</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>关闭机器臂</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>30</y>
        <width>191</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>平面间的夹角：</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="plane_angle_txt">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>30</y>
        <width>221</width>
        <height>41</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>830</x>
        <y>20</y>
        <width>181</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string> 平面的三个点：</string>
      </property>
     </widget>
     <widget class="QPushButton" name="newplane_btn">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>130</y>
        <width>171</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>平面对刀</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="newplaneangle_txt">
      <property name="geometry">
       <rect>
        <x>640</x>
        <y>130</y>
        <width>181</width>
        <height>51</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>130</y>
        <width>171</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>新平面夹角：</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>340</y>
        <width>231</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>点到平面的距离：</string>
      </property>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>1030</x>
        <y>20</y>
        <width>171</width>
        <height>97</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLineEdit" name="lineEdit_3"/>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_4"/>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_5"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>470</y>
        <width>71</width>
        <height>171</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3"/>
     </widget>
     <widget class="QPushButton" name="freedrive_btn_2">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>270</y>
        <width>121</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>自带拖动</string>
      </property>
     </widget>
     <widget class="QPushButton" name="freedrive_close_btn_2">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>270</y>
        <width>161</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>自带拖动关闭</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="pt1_dis_txt">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>380</y>
        <width>289</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>26</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </widget>
     <widget class="QLineEdit" name="pt2_dis_txt">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>500</y>
        <width>289</width>
        <height>81</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>26</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </widget>
     <widget class="QLineEdit" name="pt3_dis_txt">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>620</y>
        <width>289</width>
        <height>91</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>26</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>圆锥保持</string>
     </attribute>
     <widget class="QPushButton" name="conekeep_btn">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>100</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>圆锥保持</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="cone_dist_txt">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>120</y>
        <width>121</width>
        <height>31</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>100</y>
        <width>241</width>
        <height>61</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>圆锥保持的半径：</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>650</x>
        <y>110</y>
        <width>21</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>15</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>M</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>跟随校正</string>
     </attribute>
     <widget class="QPushButton" name="initP_btn">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>120</y>
        <width>141</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string>运动至起始位</string>
      </property>
     </widget>
     <widget class="QPushButton" name="base_clockwise_btn">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>250</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>基座顺时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="base_counterclockwise_btn">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>380</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>基座逆时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="base_counterclockwise_btn_2">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>250</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>末端Z轴顺时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Z_clockwise_btn">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>380</y>
        <width>181</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>末端Z轴逆时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Rx_clockwise_btn">
      <property name="geometry">
       <rect>
        <x>800</x>
        <y>250</y>
        <width>151</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>末端X轴顺时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Rx_counterclockwise_btn">
      <property name="geometry">
       <rect>
        <x>800</x>
        <y>380</y>
        <width>161</width>
        <height>81</height>
       </rect>
      </property>
      <property name="text">
       <string>末端X轴逆时针</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>540</x>
        <y>90</y>
        <width>211</width>
        <height>71</height>
       </rect>
      </property>
      <property name="text">
       <string> 极速测试</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="sizeConstraint">
      <enum>QLayout::SetNoConstraint</enum>
     </property>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionInit">
   <property name="text">
    <string>初始化</string>
   </property>
   <property name="toolTip">
    <string>初始化</string>
   </property>
  </action>
  <action name="actionMove">
   <property name="text">
    <string>移动</string>
   </property>
   <property name="toolTip">
    <string>移动</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
